src/dc_msg.rs

```diff
@@ -831,98 +832,74 @@ pub unsafe fn dc_msg_get_summarytext(
     type_0: Viewtype,
-    text: *const libc::c_char,
+    text: &str,
     param: &mut Params,
     approx_characters: libc::c_int,
     context: &Context,
-) -> *mut libc::c_char {
-    /* get a summary text, result must be free()'d, never returns NULL. */
-    let mut ret;
-    let mut prefix: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut pathNfilename: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut value: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut append_text: libc::c_int = 1i32;
+) -> String {
+    let ret: String;
+    let mut prefix = "".to_string();
+    let mut append_text = true;
     match type_0 {
-        Viewtype::Image => prefix = context.stock_str(StockMessage::Image).strdup(),
-        Viewtype::Gif => prefix = context.stock_str(StockMessage::Gif).strdup(),
-        Viewtype::Video => prefix = context.stock_str(StockMessage::Video).strdup(),
-        Viewtype::Voice => prefix = context.stock_str(StockMessage::VoiceMessage).strdup(),
+        Viewtype::Image => prefix = context.stock_str(StockMessage::Image).to_string(),
+        Viewtype::Gif => prefix = context.stock_str(StockMessage::Gif).to_string(),
+        Viewtype::Video => prefix = context.stock_str(StockMessage::Video).to_string(),
+        Viewtype::Voice => prefix = context.stock_str(StockMessage::VoiceMessage).to_string(),
         Viewtype::Audio | Viewtype::File => {
             if param.get_int(Param::Cmd) == Some(6) {
-                prefix = context.stock_str(StockMessage::AcSetupMsgSubject).strdup();
-                append_text = 0i32
+                prefix = context
+                    .stock_str(StockMessage::AcSetupMsgSubject)
+                    .to_string();
+                append_text = false
             } else {
-                pathNfilename = param
-                    .get(Param::File)
-                    .unwrap_or_else(|| "ErrFilename")
-                    .strdup();
-                value = dc_get_filename(pathNfilename);
-                let label = CString::new(
-                    context
-                        .stock_str(if type_0 == Viewtype::Audio {
-                            StockMessage::Audio
-                        } else {
-                            StockMessage::File
-                        })
-                        .as_ref(),
-                )
-                .unwrap();
-                prefix = dc_mprintf(
-                    b"%s \xe2\x80\x93 %s\x00" as *const u8 as *const libc::c_char,
-                    label.as_ptr(),
-                    value,
-                )
+                let value;
+                unsafe {
+                    let pathNfilename = param
+                        .get(Param::File)
+                        .unwrap_or_else(|| "ErrFilename")
+                        .strdup();
+                    value = as_str(dc_get_filename(pathNfilename));
+                    free(pathNfilename as *mut libc::c_void);
+                }
```

Comment:
I think value is a string slice pointing to something which you're then freeing, which won't end well.
```
let value = unsafe {
    let base_fname = CString::yolo(param.get(Param::File).unwrap_or("ErrFilename"));
    to_string(dc_get_filename(base_fname))
}